cpp/include/libxtreemfs/file_info.h

```diff
@@ -211,9 +184,40 @@ class FileInfo {
       bool* wait_completed,
       boost::mutex* wait_completed_mutex);
 
-  /** Get the mutex, that can be used for xLocSet renewals from multiple FileHandles. */
-  inline boost::mutex& xlocset_renewal_mutex() {
-    return xlocset_renewal_mutex_;
+
+  void UpdateXLocSetAndRest(const xtreemfs::pbrpc::XLocSet& new_xlocset,
+                                   bool replicate_on_close);
+
+  void UpdateXLocSetAndRest(const xtreemfs::pbrpc::XLocSet& new_xlocset);
+
+  /** Copies the XlocSet into new_xlocset. */
+  void GetXLocSet(xtreemfs::pbrpc::XLocSet* new_xlocset);
+
+  /** Copies the XlocSet into new_xlocset
+   *  and returns the corresponding UUIDContainer.
+   *  The UUIDcontainer is just valid for the associated XLocSet.
+   */
+  boost::shared_ptr<UUIDContainer> GetXLocSetAndUUIDContainer(
```

Comment:
@kleingeist Output-parameters are not bad practise, logically it is a call-by-reference semantic. You do not modify the pointer but the pointee. Since the new_xlocset is pre-allocated by the caller, I see no better way here.
Maybe it would be clearer to have the GetXLocSet() and GetUUIDContainer() and call both of them. But only if a) the access to UUIDContainer does not have to happen inside a locked scope, and b) copying the XLocSet and returning the UUIDContainer do have to happen atomically (within the same locked scope).